dev/image-guidelines.md

```diff
@@ -0,0 +1,335 @@
+
+Ideally we want screenshots in the documentation to have a comfortably low resolution.
+This can be achieved by either:
+
+- only showing the part of the screen that the user needs to see.
+- using a constrained screen size for full-screen screenshots.
+
+These are explained below.
+
+#### Partial screens
+
+Screenshots that show only part of the screen area are relatively straightforward.
+
+![](image-guidelines/screen_partial.png)
+
+A useful trick in Chrome Dev Tools is that you can right-click on a DOM element and capture an image of just that element.
+![](image-guidelines/node_screenshot.png)
+
+#### Full screens
+
+When full screenshots are needed we can improve their legibility by simulating a smaller screen size.
+The "dev tools" of modern browsers allows you to simulate arbitrary screen sizes, including many common phone screens.
+Velociraptor's GUI is not a fully responsive web application but it's layout remains usable at quite small (by modern standards) screen resolutions.
+
+For full screenshots use one of the following 2 sizes:
+
+- 1280x960 (preferred)
+- 1600x1200 (if needed to show wide table content)
+
+![](image-guidelines/screen_size.png)
+*These particular constrained sizes can be configured as presets in Chrome's "dev tools" for quick access.*
+
+![](image-guidelines/fullscreen.png)
+
+### Velociraptor theme
+
+For consistency, and to avoid confusing new users, we use a single GUI theme for all screenshots: the **Velociraptor Light** theme which is the default for new installations.
+
+#### Theme tweaks
+
+While screenshots do need to depict the actual application UI, it's reasonable to tweak some aspects to make the screenshots a bit clearer for documentation purposes.
+These may include:
+1. using a slightly heavier font weight: 500 rather than 400. On Windows the fonts seem to render slightly lighter for some unknown reason.
+2. using pure black or white as the text color instead of lower contrast tones.
+3. reducing the opacity of the modal backdrop so that items in the background are not excessively dimmed
+4. deleting misbehaving or obscuring elements from the DOM if necessary.
+5. adding a glow effect via CSS to focus the reader on a specific element.
+
+Points 1-3 can be patched into the GUI's CSS and compiled to make a documentation-specific binary. A patch is provided in this folder named `documentation_tweaks_{commit_id}.patch`
+
+A glow effect (point 5) around GUI elements can be achieved by applying this CSS to the DOM elements in "dev tools".
+
+Background color may be turned off if it isn't needed or produces a bad result. It's purpose is to fill in rounded corners behind the element being highlighted.
+
+Z-index is only sometimes needed for elements such as buttons when an adjacent element has a z-index property set that causes it to overlay the glow. This property ensures that the glow effect is at the top.
+
+```css
+box-shadow: 0px 0px 10px 5px rgba(34,139,230,0.9);
```

Comment:
Although I think it looks cool I dont think it works too well on all elements and style:

With the background-color it doesnt quite look like your screenshot
![image](https://github.com/user-attachments/assets/b7a7c042-3cd7-47ef-99c9-27f7bd8b1d99)

Without the background color it looks closer but messes up some icons
![image](https://github.com/user-attachments/assets/78c44bf2-31ce-473d-bbb2-21e946d1431e)

We also need to be careful to not place the highlight on the td as it basically messes it right up
![image](https://github.com/user-attachments/assets/82624dae-2f6d-4641-a7a7-f9dcea3240ba)
The box shadow does not work on td and only the background color covers the whole thing.

I am also a bit concerned with the level of effort this requires to update screenshots - if we do the highlighting in the drawing program it is just a matter of replacing the underlying png layer and aligning with existing arrows / text,  but now we have to go through all the elements to re-highlight them before we take the screenshot which is more work.

Simple circles and squares are probably good enough and take less effort 
![image](https://github.com/user-attachments/assets/f755bb94-974c-489d-a455-e0d05ee8bf34)

Reply:
Sure this was just an idea. I've updated the doc to use drawn shapes instead.